### ISSUE TITLE
feature request: 3D version with different algorithm

### ISSUE THREAD

**Issue by magnetophon** [marierm/mmExtensions#3]
I'm not expecting you to go out and make this, but I wanted to trow in the idea anyway:

I would like a version that is limited to 3 maximum dimensions (for use with a kinekt), that uses a reverse distance algorithm (I think that's what it is called).
What I mean is: when the cursor is on a preset, then only that preset should determine the output.

I realize that for your use case, with more dimensions, you need the current algorithm, and it's a very clever one too.

Related question: how do you personally deal with presets never being presented "pure"?
Say I'm controlling an FM synth, and I want to go from pure sine to harsh harmonics.
In the current situation, you can only approximate that by putting a couple of sine sounding presets close to each other on one end of the range, and a few harsh ones on the other.

Sorry for the rant...  :)

**Comment by marierm** [comment]
It has been a while.  Sorry I missed that post.

I do not understand why you would need a version that's limited to three dimensions.  You can use the interpolator with the number of dimensions you want, including three.
A reverse distance algorithm would work too, but the current algorithm is _exact_ as well, meaning that, when the cursor is positioned exactly on a preset, only that preset determines the output.  I feel like the current algorithm is better because it is _local_, meaning that not all points are taken account when doing the interpolation (just the nearest ones).  When using a reverse distance algo, all points are taken into account all the time and distant points may have an undesirable impact on the result.

Regarding your last question, I'd say that it is very important for me to be able to access the "pure" presets, and I designed an algo that is _exact_ for that reason.

It did not feel like a rant to me.  Thanks for your interest.  :-)

**Comment by magnetophon** [comment]
Did the algorithm change in the last year?
I don't have a way to test now, but I was convinced you couldn't have pure presets when I wrote that issue.

I'll test again.

**Comment by marierm** [comment]
No, the algo did not change at all.  If you find that the interpolation is not exact, I would very much like to know what the context is.  Please post some code if you have time.  Thank you!
